### ISSUE TITLE
Questions about Custom training for kws_streaming.

### ISSUE THREAD

**Issue by utkarsh-21st** [google-research/google-research#579]
Firstly, a big thanks for open sourcing this work.

I am following [01_train.ipynb](https://github.com/google-research/google-research/blob/master/kws_streaming/colab/01_train.ipynbl) and [02_inference.ipynb](https://github.com/google-research/google-research/blob/master/kws_streaming/colab/02_inference.ipynbl) for training a model for keyword detection on a dataset which has only 1 class (+ unknown + silence).
I have 2 questions:
1. What size of the dataset will I require to train a model that is robust enough to be relied upon in real life? (are 3000 instances of keyword class of 1-sec audio clips good enough?) (size of unknown words?)

2. Tips, suggestions for training?

**Comment by rybakov** [comment]
kws_streaming was designed as a prototype tool for keras streaming aware models.
In the paper we use accuracy metrics and data sets because of:
* accuracy metrics were selected only for the purpose of comparison with previously published work. for real life you should extend it with False accept rate and miss rate, also measure some of these metrics in time: false accept rate per hour or something like this.
* we selected speech commands data for benchmarking and reporting purpose and comparison with previously published papers. But in real applications you should have more data with different accent, gender, age categories plus different kind of noise environments (indoor, outdoor, car, etc).

So it is hard to say how much data you need without considering the context of the application.
